cluster/cluster.go

```diff
@@ -386,6 +425,19 @@ func (cluster *Cluster) GenerateAndExecuteCommand(verboseMsg string, scope Scope
 }
 
 func (cluster *Cluster) CheckClusterError(remoteOutput *RemoteOutput, finalErrMsg string, messageFunc interface{}, noFatal ...bool) {
+	for _, retriedCommand := range remoteOutput.RetriedCommands {
+		switch messageFunc.(type) {
+		case func(content int) string:
+			content := retriedCommand.Content
+			host := cluster.GetHostForContent(content)
+			gplog.Debug("Command failed before passing on segment %d on host %s with error:\n%v", content, host, retriedCommand.RetryError)
```

Comment:
If a command fails until the end of maxAttempts it will end up in the failedCommands list rather than the retriedCommands list. Commands will only end up in one of those two lists. So any command at this stage was retried but eventually passed.